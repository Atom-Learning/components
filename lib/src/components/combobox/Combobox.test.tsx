import { fireEvent, render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

import { Combobox } from '.'

const SimpleExample = () => (
  <Combobox openOnFocus aria-label="Favourite fruit">
    <Combobox.Input id="fruit" />
    <Combobox.Popover>
      <Combobox.List>
        <Combobox.Option value="Apple" />
        <Combobox.Option value="Banana" />
        <Combobox.Option value="Cranberry" />
        <Combobox.Option value="Dragon fruit" />
      </Combobox.List>
    </Combobox.Popover>
  </Combobox>
)

describe('Combobox component', () => {
  it('renders', async () => {
    const { container } = render(<SimpleExample />)

    expect(container).toMatchSnapshot()
  })

  it('has no programmatically detectable a11y issues', async () => {
    const { container } = render(<SimpleExample />)

    expect(await axe(container)).toHaveNoViolations()
  })

  it('receives value from list on selection', async () => {
    render(<SimpleExample />)

    const combobox = screen.getByRole('combobox')

    fireEvent.focus(combobox)
    fireEvent.click(screen.getByText('Apple'))

    expect(combobox).toHaveValue('Apple')
  })

  it('renders with lg size', async () => {
    const { container } = render(
      <Combobox openOnFocus aria-label="Favourite fruit">
        <Combobox.Input id="fruit" size="lg" />
        <Combobox.Popover>
          <Combobox.List>
            <Combobox.Option value="Apple" />
          </Combobox.List>
        </Combobox.Popover>
      </Combobox>
    )

    expect(container).toMatchSnapshot()
  })
})
