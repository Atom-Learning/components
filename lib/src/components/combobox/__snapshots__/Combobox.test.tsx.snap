// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox component renders 1`] = `
@media  {
  .c-bYKHLS {
    box-shadow: none;
    font-size: var(--fontSizes-md);
    -webkit-appearance: none;
    appearance: none;
    background-image: url(data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224px%22%20height%3D%2224px%22%20viewBox%3D%220%200%2024%2024%22%20stroke%3D%22%23757575%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20fill%3D%22none%22%20color%3D%22%23757575%22%3E%3Cpolyline%20points%3D%226%2010%2012%2016%2018%2010%22%2F%3E%3C%2Fsvg%3E);
    background-position: right var(--space-3) top 50%, 0 0;
    background-repeat: no-repeat, repeat;
    background-size: 20px auto, 100%;
    border: 1px solid var(--colors-tonal300);
    border-radius: var(--radii-0);
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    height: var(--sizes-4);
    padding-left: var(--space-3);
    padding-right: var(--space-6);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-bYKHLS::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }

  .c-bYKHLS:focus-within {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-bYKHLS[disabled] {
    background-color: var(--colors-tonal100);
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }

  .c-jJeEQg {
    background: white;
    border: solid 1px var(--colors-tonal100);
    border-radius: var(--radii-0);
    box-shadow: var(--shadows-1);
    box-sizing: border-box;
    font-family: var(--fonts-body);
    font-size: var(--fontSizes-md);
    outline: none;
    padding: var(--space-1);
    transform: translateY(var(--space-2));
  }

  .c-uYIcy {
    color: var(--colors-tonal500);
    cursor: pointer;
    margin: 0;
    padding: var(--space-2);
  }

  .c-uYIcy:hover,
  .c-uYIcy[aria-selected="true"] {
    background: var(--colors-tonal50);
    border-radius: var(--radii-0);
  }

  .c-uYIcy [data-user-value] {
    color: var(--colors-primary);
  }

  .c-cOrACU {
    list-style: none;
    margin: 0;
    padding: 0;
    -webkit-user-select: none;
    user-select: none;
  }
}

@media  {
  .c-bYKHLS-fwoCmE-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }
}

<div>
  <div
    class="c-PJLV"
    data-reach-combobox=""
    data-state="idle"
  >
    <input
      aria-autocomplete="both"
      aria-controls="listbox--1"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Favourite fruit"
      class="c-bYKHLS c-bYKHLS-fwoCmE-size-md"
      data-reach-combobox-input=""
      data-state="idle"
      id="fruit"
      role="combobox"
      value=""
    />
  </div>
</div>
`;

exports[`Combobox component renders with lg size 1`] = `
@media  {
  .c-bYKHLS {
    box-shadow: none;
    font-size: var(--fontSizes-md);
    -webkit-appearance: none;
    appearance: none;
    background-image: url(data:image/svg+xml;charset=US-ASCII,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2224px%22%20height%3D%2224px%22%20viewBox%3D%220%200%2024%2024%22%20stroke%3D%22%23757575%22%20stroke-width%3D%222%22%20stroke-linecap%3D%22round%22%20stroke-linejoin%3D%22round%22%20fill%3D%22none%22%20color%3D%22%23757575%22%3E%3Cpolyline%20points%3D%226%2010%2012%2016%2018%2010%22%2F%3E%3C%2Fsvg%3E);
    background-position: right var(--space-3) top 50%, 0 0;
    background-repeat: no-repeat, repeat;
    background-size: 20px auto, 100%;
    border: 1px solid var(--colors-tonal300);
    border-radius: var(--radii-0);
    box-sizing: border-box;
    color: var(--colors-tonal600);
    cursor: text;
    display: block;
    font-family: var(--fonts-body);
    height: var(--sizes-4);
    padding-left: var(--space-3);
    padding-right: var(--space-6);
    transition: all 100ms ease-out;
    width: 100%;
  }

  .c-bYKHLS::placeholder {
    color: var(--colors-tonal300);
    opacity: 1;
  }

  .c-bYKHLS:focus-within {
    border-color: var(--colors-primary);
    outline: none;
  }

  .c-bYKHLS[disabled] {
    background-color: var(--colors-tonal100);
    color: var(--colors-tonal400);
    cursor: not-allowed;
  }

  .c-jJeEQg {
    background: white;
    border: solid 1px var(--colors-tonal100);
    border-radius: var(--radii-0);
    box-shadow: var(--shadows-1);
    box-sizing: border-box;
    font-family: var(--fonts-body);
    font-size: var(--fontSizes-md);
    outline: none;
    padding: var(--space-1);
    transform: translateY(var(--space-2));
  }

  .c-uYIcy {
    color: var(--colors-tonal500);
    cursor: pointer;
    margin: 0;
    padding: var(--space-2);
  }

  .c-uYIcy:hover,
  .c-uYIcy[aria-selected="true"] {
    background: var(--colors-tonal50);
    border-radius: var(--radii-0);
  }

  .c-uYIcy [data-user-value] {
    color: var(--colors-primary);
  }

  .c-cOrACU {
    list-style: none;
    margin: 0;
    padding: 0;
    -webkit-user-select: none;
    user-select: none;
  }
}

@media  {
  .c-bYKHLS-fwoCmE-size-md {
    height: var(--sizes-4);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }

  .c-bYKHLS-jAZHIx-size-lg {
    height: var(--sizes-5);
    font-size: var(--fontSizes-md);
    line-height: 2;
  }
}

<div>
  <div
    class="c-PJLV"
    data-reach-combobox=""
    data-state="idle"
  >
    <input
      aria-autocomplete="both"
      aria-controls="listbox--13"
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-label="Favourite fruit"
      class="c-bYKHLS c-bYKHLS-jAZHIx-size-lg"
      data-reach-combobox-input=""
      data-state="idle"
      id="fruit"
      role="combobox"
      value=""
    />
  </div>
</div>
`;
